packages/workspace/src/parser/internal/native/lexer.ts

```
@@ -110,6 +124,14 @@ class PeekableLexer {
     private advance(): LexerToken | undefined {
       let token = this.lexer.next()
 
+      if (token?.type === TOKEN_TYPES.MERGE_CONFLICT) {
```

this should definitely not be here, it would not validate the actual next token (according to the rest of the code, the token being returned isn't this one)
looks like there is already a place for `validateToken`, why is this not there?

---

+1
This should be a part of the validate token function

---

Just make sure to throw instead of returning undefined :)

---

Moved it to validateToken function